service/matching/fx.go

```diff
@@ -110,6 +111,21 @@ func ServiceResolverProvider(membershipMonitor membership.Monitor) (membership.S
 	return membershipMonitor.GetResolver(primitives.MatchingService)
 }
 
+// This type alias is used to ensure the replicator only gets set if global namespaces are enabled on this cluster.
```

Comment:
technically this is not a "type alias", it's a new type. you can just say:
```suggestion
// This type is used to ensure the replicator only gets set if global namespaces are enabled on this cluster.
```